### PR TITLE
backport of multiple fixes

### PR DESCRIPTION
**Pull request by jiaqiluo**
New Issue | Original PR | Description | Supporting PR
------ | --------------- | ----------- | ------
https://github.com/rancher/rancher/issues/34466 | https://github.com/rancher/backup-restore-operator/pull/149 | New way to handle subResource Status | https://github.com/rancher/rancher/pull/34321
https://github.com/rancher/rancher/issues/34467 | https://github.com/rancher/backup-restore-operator/pull/147 | remove the deployment rancher from the `resouceSet`
https://github.com/rancher/rancher/issues/34553| https://github.com/rancher/backup-restore-operator/pull/151 | support both `apiextensions.k8s.io/v1beta1` and `apiextensions.k8s.io/v1`

TODO after merging 

- [ ] cut new tag

### CODE REVIEW

**Review by snasovich — APPROVED** [review]
LGTM, effectively the same as all referenced PRs merged

**Review by nickgerace — APPROVED** [review]

**Comment by jiaqiluo** [comment]
The backport PR can be merged after two Devs' approval.

**Pull request merged**
